src/main/java/org/jenkinsci/plugins/codesonar/services/HubVersionService.java

```diff
@@ -0,0 +1,181 @@
+package org.jenkinsci.plugins.codesonar.services;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.http.Consts;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.ParseException;
+import org.apache.http.client.fluent.Request;
+import org.apache.http.util.EntityUtils;
+import org.jenkinsci.plugins.codesonar.CodeSonarPublisher;
+import org.jenkinsci.plugins.codesonar.models.CodeSonarHubInfo;
+import org.jenkinsci.plugins.codesonar.models.VersionCompatibilityInfo;
+
+import com.google.common.base.Throwables;
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+
+import hudson.AbortException;
+
+public class HubVersionService {
+    private static final Logger LOGGER = Logger.getLogger(HubVersionService.class.getName());
+    
+    private HttpService httpService;
+
+    public HubVersionService(HttpService httpService) {
+        this.httpService = httpService;
+    }
+
+    public CodeSonarHubInfo getHubVersion(URI baseHubUri) throws AbortException {
+        LOGGER.log(Level.INFO, String.format("Retrieving CodeSonar hub version"));
+        
+        CodeSonarHubInfo hubVersion = new CodeSonarHubInfo();
+        
+        VersionCompatibilityInfo vci = fetchVersionCompatibilityInfo(baseHubUri, CodeSonarPublisher.CODESONAR_HUB_CLIENT_NAME, CodeSonarPublisher.CODESONAR_HUB_CLIENT_PROTOCOL_VERSION_NUMBER);
+            
+        if(vci != null) {
+            hubVersion.setVersion(vci.getHubVersion());
+            
+            //If this client is supposed to be able to talk to the hub
+            if(checkClientOk(vci)) {
+                hubVersion.setOpenAPISupported(supportsOpenAPI(vci));
+            } else {
+                //In this case this client has been rejected by the hub
+                throw new AbortException(String.format("[CodeSonar] client rejected by the hub. %n[CodeSonar] clientOK=%s", vci.getClientOK().toString()));
+            }
+        } else {
+            /*
+             * In such a case the hub version is most likely to be <= 7.0, as a result
+             * let's try loading the hub version through the legacy endpoint (/command/anon_info/)
+             */
+            LOGGER.log(Level.INFO, String.format("Falling back to legacy version endpoint (/command/anon_info/)"));
+            String hubVersionLegacy = fetchHubVersionLegacy(baseHubUri);
+            hubVersion.setVersion(hubVersionLegacy);
```

Comment:
I don't think it is accurate to call this the "legacy endpoint".  The purpose of "anon_info" is to show the CodeSonar hub server version "signature".  The purpose of "check_version" is to check the compatibility of a CodeSonar hub client.  The "anon_info" endpoint is used here as a fallback, not because it is "legacy", but because "check_version" is not available.

I think the comment should say:
> we cannot ask the hub for capabilities and compatibility, but we can try to extract the hub's version from different endpoint.

I think the log comment should not mention "anon_info" since the fact that you are using "anon_info" is an implementation detail of the `fetchHubVersionLegacy()` method.  In fact, I think you should name that method something more specific that doesn't say "legacy", for example `fetchHubSignatureVersionString()".
Instead of "Falling back to legacy version endpoint...",I think the log message should be:
> "Hub did not provide client compatibility information.  Attempting to query hub version signature..."